source/fuzz/transformation_equation_instruction.cpp

```diff
@@ -131,51 +164,150 @@ uint32_t TransformationEquationInstruction::MaybeGetResultType(
+    rhs_id.push_back(id);
+  }
+
+  FindInstruction(message_.instruction_to_insert_before(), ir_context)
+      ->InsertBefore(MakeUnique<opt::Instruction>(
+          ir_context, static_cast<SpvOp>(message_.opcode()),
+          ComputeResultTypeId(ir_context), message_.fresh_id(),
+          std::move(in_operands)));
+
+  ir_context->InvalidateAnalysesExceptFor(opt::IRContext::kAnalysisNone);
+
+  transformation_context->GetFactManager()->AddFactIdEquation(
+      message_.fresh_id(), static_cast<SpvOp>(message_.opcode()), rhs_id,
+      ir_context);
+}
+
+protobufs::Transformation TransformationEquationInstruction::ToMessage() const {
+  protobufs::Transformation result;
+  *result.mutable_equation_instruction() = message_;
+  return result;
+}
+
+uint32_t TransformationEquationInstruction::ComputeResultTypeId(
+    opt::IRContext* ir_context) const {
+  auto opcode = static_cast<SpvOp>(message_.opcode());
+  switch (opcode) {
+    case SpvOpConvertUToF:
+    case SpvOpConvertSToF: {
+      assert(message_.in_operand_id_size() == 1 &&
+             "Instruction has invalid number of operands");
+
+      const auto* type = ir_context->get_type_mgr()->GetType(
+          fuzzerutil::GetTypeId(ir_context, message_.in_operand_id(0)));
+      assert(type && "Operand has invalid type");
+
+      if (const auto* vector = type->AsVector()) {
+        assert(vector->element_type()->AsInteger() &&
+               "Conversion to float supports only operands of scalar or vector "
+               "integral type");
+
+        return fuzzerutil::MaybeGetVectorType(
+            ir_context,
+            fuzzerutil::MaybeGetFloatType(
+                ir_context, vector->element_type()->AsInteger()->width()),
+            vector->element_count());
+      } else {
+        assert(type->AsInteger() &&
+               "Conversion to float supports only operands of scalar or vector "
+               "integral type");
+
+        return fuzzerutil::MaybeGetFloatType(ir_context,
+                                             type->AsInteger()->width());
       }
-      return operand_inst->type_id();
     }
+    case SpvOpConvertFToU:
+    case SpvOpConvertFToS: {
+      assert(message_.in_operand_id_size() == 1 &&
+             "Instruction has invalid number of operands");
+
+      const auto* type = ir_context->get_type_mgr()->GetType(
+          fuzzerutil::GetTypeId(ir_context, message_.in_operand_id(0)));
+      assert(type && "Operand has invalid type");
 
+      if (const auto* vector = type->AsVector()) {
+        assert(vector->element_type()->AsFloat() &&
+               "Conversion to integer supports only operands of scalar or "
+               "vector float type");
+
+        return fuzzerutil::MaybeGetVectorType(
+            ir_context,
+            fuzzerutil::MaybeGetIntegerType(
+                ir_context, vector->element_type()->AsFloat()->width(),
+                opcode == SpvOpConvertFToS),
```

Comment:
Again, it would be good if this could be more relaxed.

Reply:
Fixed in a6ad6b3.